client/src/components/Datalist/index.js

```diff
@@ -4,17 +4,19 @@ import { uniq } from "lodash";
 class Datalist extends React.PureComponent {
   render() {
     let options = [];
-
     if (this.props.options && this.props.processOptions) {
-      options = this.props.processOptions(this.props.options);
+      options = this.props.processOptions(
```

Comment:
Datalist should remain agnostic about the way this.props.processOptions works.  It either gets 

1) no processOptions prop passed to it, in which case it just uses the props.options that was passed to it
 or 
2) a processOptions prop is passed and the props.options are passed through that function to arrive at a set of usable options.  

It's up to the caller to ensure that their `processOptions` call can process the `options` they passed in a way that makes sense.  So in this case, rather than changing the way this component works and saying it could get `optionalArgs`, you're just going to make sure that the `options` prop you pass is a set that includes what you're calling `options` and `optionalArgs` below.  

You don't want this component to care at all about how `processOptions` works or how many args it needs.  That's the responsibility of the caller.